### PR TITLE
[otbn] Remove reset from FF-based GPRs and WDRs

### PR DESCRIPTION
**Pull request by andreaskurth**
~**PR Dependency**: This PR depends on #13724 and currently contains duplicate commits from that PR. Please only review the last commit in this PR.~

This removes the reset from the flip-flop-based general purpose registers (GPRs) and wide data registers (WDRs).

This improves security, as [argued by @vogelpi](https://github.com/lowRISC/opentitan/pull/12741#discussion_r893497686):
> [The] OTBN register files are a good example for the use of non-reset flops. The main problem is not the leakage occurring while resetting the register. We are more worried about what's happening afterwards. For example if a register holds a random mask to be applied to some other register (remember that the masking is implemented in SW here) and an attacker manages to reset that register while OTBN is running, the masking is effectively switched off. I believe the integrity bits don't really help here because the reset line is also connected to these. Synthesis will just insert set instead of reset flops for these bits. I am proposing to switch to non-reset flops for the register files and the ISPRs + RND/URND paths (incl. RND cache).

This is part of the work on OTBN D2S (#11019).

### CODE REVIEW

**Comment by andreaskurth** [comment]
As is, this causes problems with uninitialized registers (described in the PR body). To prevent those problems, we have decided to let OTBN run an initial secure wipe when it comes out of reset. This is being implemented in #13321. Once #13321 is merged, this PR will be continued by rebasing and removing the SW changes.

**Comment by tjaychen** [comment]
following on from my comment [here](https://github.com/lowRISC/opentitan/pull/12741#discussion_r896971028)
we might want to think about whether the GPR / WDR is also used to hold actual secrets (like a seed / key).  Adding @jadephilipoom to comment. 

I think even if it were, it still makes sense to do this since the bar for getting into "scan" mode while in a production state is [pretty high](https://docs.google.com/document/d/1PNyfKBteoVfaex-zPNg5nY8ny9hyeHII9eJvgK23gjI/edit#heading=h.11pq5qd49n24).  And if somewhere were to do that, we have much more serious problems anyways. 

So this still feels like the right trade-off to me... @cdgori / @johannheyszl what do you guys think?

**Comment by jadephilipoom** [comment]
This change makes sense to me; from a software perspective, I don't really see a good legitimate reason to read from an uninitialized register, so if it's happening it likely indicates an attack.

**Comment by johannheyszl** [comment]
thx for the discussion @vogelpi, and summarizing it here:

tl/dr: lgtm

Intro: FI on reset net drivers could set OTBN regs to zero value, which (unfortunately) is valid in respect to the OTBN integrity check mechanisms of those regs. Many other FIs on FFs of those regs would be detected through the same integrity checks. 

Reasons to remove such potentially FI-prone reset nets:
(1) I agree that removing RST from FFs is helpful to prevent combined (FI+SCA) attacks where after FI on reset of a reg, SCA leakage occurs in this same moment in time due to zeroing, but also may zeroe-out masking values held in such regs. Even though a combined FI+SCA attack requires more attacker resources (precise FI as well as SCA, hence, two setups), this seems reasonable and worth it. 
However, this is not only an SCA issue IMO.
(2) Zeroing entire regs while circumventing integrity protection during asymm. crypto operations could result in crypto attacks based on outputs (e.g. when zeroing parts of a scalar in ECC based on signatures). This seems even worse to me bc such an attack would not require an SCA setup, ‘only’ an FI setup and access to outputs. @jadephilipoom pls correct me if I am wrong here.
-> Summary: there are two good reasons to remove it.

@tjaychen comment re scan issue seems reasonable so no red flag there.

Importantly, I also agree that the secure wipe is mandatory for this (to prevent SW attacks). This has important properties:
- As far as I understand from [#13321](https://github.com/lowRISC/opentitan/pull/13321) there shall be no way for SW to abort secure wipe.  This is absolutely necessary.
- It is very important that secure wipe uses ‘rnd’ values (non-deterministic, non-constant for attacker), otherwise we would be giving attackers a serialized word-wise hamming distance to known/constant values during wiping. 
I may be helpful to allow for a SW triggered secure wipe (e.g. to clean up after crypto operations).

**Review by johannheyszl — APPROVED** [review]
lgtm as explained in comment

**Review by GregAC — APPROVED** [review]

**Comment by vogelpi** [comment]
#13724 has been merged and I've rebased this accordingly. Will merge once CI passes.
